Assets/_Functionalities/OGC3DTiles/Scripts/Tile3DLayerGameObject.cs

```diff
@@ -62,14 +60,45 @@ private void EnableTileset()
         protected void Awake()
         {
             tileSet = GetComponent<Read3DTileset>();
+            
+            credentialHandler = GetComponent<ICredentialHandler>();
+            if(!string.IsNullOrEmpty(tileSet.tilesetUrl))
+                credentialHandler.BaseUri = new Uri(tileSet.tilesetUrl); //apply the URL from what is serialized in the tileset component.
+            
+            credentialHandler.OnAuthorizationHandled.AddListener(HandleCredentials);
             urlPropertyData = new Tile3DLayerPropertyData(TilesetURLWithoutQuery(tileSet.tilesetUrl));
+            //listen to property changes in start and OnDestroy because the object should still update its transform even when disabled
+            urlPropertyData.OnUrlChanged.AddListener(UpdateURL);
 
             if (usePropertySections)
                 propertySections = GetComponents<IPropertySectionInstantiator>().ToList();
             else
                 propertySections = new();
         }
 
+        private void HandleCredentials(StoredAuthorization auth)
+        {
+            ClearCredentials();
+            
+            if (auth is BearerToken bearerToken) //todo: moet BearerToken inheriten van InferableSingle key of niet?
+            {
+                tileSet.AddCustomHeader(bearerToken.headerName, bearerToken.headerPrefix + bearerToken.key);
+                tileSet.RefreshTiles();
+            }
+            else if (auth is QueryStringAuthorization inferableSingleKey)
+            {
+                tileSet.personalKey = inferableSingleKey.key;
+                tileSet.publicKey = inferableSingleKey.key;
+                tileSet.QueryKeyName = inferableSingleKey.queryKeyName;
+                tileSet.RefreshTiles();
+            }
+            else if (auth is UsernamePassword usernamePassword)
+            {
+                tileSet.AddCustomHeader(usernamePassword.headerName, usernamePassword.headerPrefix + usernamePassword.GetUserNamePassWordQuery(), true);
```

Comment:
Als je dit veranderd zodat de HeaderBasedAuthorization een HeaderValue property of method heeft, dan encapsuleer je de opbouw waar die hoort - en dan hoef je hier alleen een check te doen of het een HeaderBasedAuthorization is en geen onderscheid tussen UsernamePassword en Bearer